src/www/status_dhcpv6_leases.php

```diff
@@ -97,15 +97,49 @@ function parse_duid($duid_string) {
 $leasesfile = services_dhcpdv6_leasesfile();
 
 if ($_SERVER['REQUEST_METHOD'] === 'GET') {
-    $awk = "/usr/bin/awk";
-
-    /* this pattern sticks comments into a single array item */
-    $cleanpattern = "'{ gsub(\"^#.*\", \"\");} { gsub(\"^server-duid.*\", \"\");} { gsub(\";$\", \"\"); print;}'";
-    /* We then split the leases file by } */
-    $splitpattern = "'BEGIN { RS=\"}\";} {for (i=1; i<=NF; i++) printf \"%s \", \$i; printf \"}\\n\";}'";
-
-    /* stuff the leases file in a proper format into a array by line */
-    exec("/bin/cat {$leasesfile} | {$awk} {$cleanpattern} | {$awk} {$splitpattern} | /usr/bin/grep '^ia-.. '", $leases_content);
+    $leases_content = array();
+
+    $t_dhcpd6_leases = file_get_contents($leasesfile);
```

Comment:
@AdSchellevis @marjohn56  It does not hurt early - there is a user who is reading a 40MiB log file which means that you have to have this in memory first. I had to do this change as well because this is almost the limit configured in PHP.

You can look at opnsense/plugins#1075 to see how I implemented this some days ago.

Reply:
I started out looking at using fread, but a single lease entry is across multiple lines, that means that an may cross the buffer boundary. Guess I could do it by copying the partial entry to a variable first, then reading more into the buffer etc etc, or I could just use read fgets and assemble thereafter.